### ISSUE TITLE
Add history for votes

### ISSUE THREAD

**Issue by flotwig** [snoonetIRC/staffvote#4]
It would be good to have history for each poll, including:

* New votes
* When a user changes their vote (to see if someone's waffling)
* When the description/title of the poll is changed

Each history mark would have a timestamp and a responsible user as well as the action which occurred.

**Comment by aFreshMelon** [comment]
I will create a logger for this. And by create I mean I will likely use a gem.

**Comment by flotwig** [comment]
A log might be too much, it could be stored more efficiently in a table like

| (int) poll ID | (int) responsible user | (int) action (0=new vote,1=changed vote, etc.) | (misc) data like old title, old vote value, etc

**Comment by aFreshMelon** [comment]
The cleanest and most modern approach would be firing events.

**Comment by n7st** [comment]
I am making a program for exactly this I think. The entire project. It's almost done. Can we talk about this tomorrow? If it's totally custom I will put in to this to regardless.

**Comment by flotwig** [comment]
@aFreshMelon Right, a log class which compresses log values in this way would be the best

@n7st "this" referring to what?

**Comment by aFreshMelon** [comment]
I don't know if rails has built in event firing/handling engine. Would need some research.

**Comment by flotwig** [comment]
¯\_(ツ)_/¯ 

Or just logger.log_event poll.id user.username (action) (data)

**Comment by aFreshMelon** [comment]
I'd rather log into a DB than into a file. Not sure whether the logger supports this.

**Comment by n7st** [comment]
A voting app. May be overkill for what we want. Will talk to you when I'm not in the pub.

**Comment by flotwig** [comment]
Alright @n7st  

Agreed @aFreshMelon
